lib/model/query/entities.js

```diff
@@ -356,8 +367,16 @@ const _updateEntity = (dataset, entityData, submissionId, submissionDef, submiss
 
 // Used by _updateVerison to figure out the intended base version in Central
 // based on the branchId, trunkVersion, and baseVersion in the submission
-const _computeBaseVersion = (eventId, dataset, clientEntity, submissionDef, forceOutOfOrderProcessing = false) => async ({ Entities }) => {
-  if (!clientEntity.def.branchId) {
+const _computeBaseVersion = (eventId, dataset, clientEntity, submissionDef, forceOutOfOrderProcessing = false, createSubAsUpdate = false) => async ({ Entities }) => {
```

Comment:
Do `forceOutOfOrderProcessing` and `createSubAsUpdate` need to be optional parameters? There are a few different methods with optional parameters at this point, and I'm having a little trouble following how each is called (with all parameters specified vs. not). If this method doesn't need optional parameters, I think it'd be a little clearer to make the parameters required.